pycbc/distributions/qnm.py

```diff
@@ -0,0 +1,260 @@
+
+    >>> dist = UniformF0Tau(f0=(10., 2048.), tau=(1e-4,1e-2),
+            final_mass=(20., 200.), final_spin=(0,0.996))
+
+    Check that all random samples drawn from the distribution are in the
+    final mass and spin constraints:
+
+    >>> samples = dist.rvs(size=1000)
+    >>> (conversions.final_mass_from_f0_tau(samples['f0'],
+            samples['tau']) >= 20.).all()
+    True
+    >>> (conversions.final_mass_from_f0_tau(samples['f0'],
+            samples['tau']) < 200.).all()
+    True
+    >>> (conversions.final_spin_from_f0_tau(samples['f0'],
+            samples['tau']) >= 0.).all()
+    True
+    >>> (conversions.final_spin_from_f0_tau(samples['f0'],
+            samples['tau']) < 0.996).all()
+    True
+
+    """
+
+    name = 'uniform_f0_tau'
+
+    def __init__(self, f0=None, tau=None, final_mass=None, final_spin=None,
+                 rdfreq='f0', damping_time='tau', norm_tolerance=1e-3,
+                 norm_seed=0):
+        if f0 is None:
+            raise ValueError("must provide a range for f0")
+        if tau is None:
+            raise ValueError("must provide a range for tau")
+        self.rdfreq = rdfreq
+        self.damping_time = damping_time
+        parent_args = {rdfreq: f0, damping_time: tau}
+        super(UniformF0Tau, self).__init__(**parent_args)
+        if final_mass is None:
+            final_mass = (0., numpy.inf)
+        if final_spin is None:
+            final_spin = (-0.996, 0.996)
+        self.final_mass_bounds = boundaries.Bounds(
+            min_bound=final_mass[0], max_bound=final_mass[1])
+        self.final_spin_bounds = boundaries.Bounds(
+            min_bound=final_spin[0], max_bound=final_spin[1])
+        # Re-normalize to account for cuts: we'll do this by just sampling
+        # a large number of spaces f0 taus, and seeing how many are in the
+        # desired range.
+        # perseve the current random state
+        s = numpy.random.get_state()
+        numpy.random.seed(norm_seed)
+        nsamples = int(1./norm_tolerance**2)
+        draws = super(UniformF0Tau, self).rvs(size=nsamples)
+        # reset the random state
+        numpy.random.set_state(s)
+        num_in = self._constraints(draws).sum()
+        # if num_in is 0, than the requested tolerance is too large
+        if num_in == 0:
+            raise ValueError("the normalization is < then the norm_tolerance; "
+                             "try again with a smaller nrom_tolerance")
+        self._lognorm += numpy.log(num_in) - numpy.log(nsamples)
+        self._norm = numpy.exp(self._lognorm)
+
+    def __contains__(self, params):
+        isin = super(UniformF0Tau, self).__contains__(params)
+        if isin:
+            isin &= self._constraints(params)
+        return isin
+
+    def _constraints(self, params):
+        f0 = params[self.rdfreq]
+        tau = params[self.damping_time]
+        # temporarily silence invalid warnings... these will just be ruled out
```

Comment:
This will get rid of all those "invalid value encountered in [...]" warning, is that correct? I'd love that!!

Reply:
I think it will... not sure.